backend/app/status_manager.py

```
@@ -0,0 +1,203 @@
+    if job_id in crawl_jobs:
+        logger.warning(f"Job ID {job_id} already exists. Re-initializing.")
+    normalized_root_url = normalize_url(root_url) # Normalize before use
+    # Assign the new status object to the managed dict
+    # Note: Pydantic models might need special handling with Manager dicts.
+    # If direct assignment fails, we might need to convert to a regular dict first.
+    try:
+        new_status = CrawlJobStatus(
+            job_id=job_id,
+            overall_status='initializing',
+            urls={normalized_root_url: 'pending_discovery'},
+            start_time=datetime.now(),
```

Ruff (0.8.2)

`datetime.datetime.now()` called without a `tz` argument

(DTZ005)